python/node_classification.py

```diff
@@ -0,0 +1,641 @@
+from torch_geometric.data import Data
+from mage.node_classification.models.inductive_model import InductiveModel
+from mage.node_classification.models.gatjk import GATJK
+from mage.node_classification.utils.metrics import metrics
+from mage.node_classification.utils.extract_from_database import extract_from_database
+from mage.node_classification.models.train_model import train_epoch
+import mgp
+from tqdm import tqdm
+import torch
+import numpy as np
+import os
+from torch_geometric.nn import to_hetero
+import typing
+
+
+##############################
+# constants
+##############################
+
+# parameters for the model
+class ModelParams:
+    IN_CHANNELS = "in_channels"
+    OUT_CHANNELS = "out_channels"
+    HIDDEN_FEATURES_SIZE = "hidden_features_size"
+    LAYER_TYPE = "layer_type"
+    AGGREGATOR = "aggregator"
+
+
+# paramters for type of layers inside model
+class LayerType:
+    gat = "GAT"
+    gatv2 = "GATv2"
+    sage = "SAGE"
+    gatjk = "GATJK"
+
+
+# parameters for optimizer
+class OptimizerParams:
+    LEARNING_RATE = "learning_rate"
+    WEIGHT_DECAY = "weight_decay"
+
+
+# parameters for data
+class DataParams:
+    SPLIT_RATIO = "split_ratio"
+    METRICS = "metrics"
+
+
+# parameters relevant to memgraph database
+class MemgraphParams:
+    NODE_ID_PROPERTY = "node_id_property"
+
+
+# parameters for training
+class TrainParams:
+    NUM_EPOCHS = "num_epochs"
+    CONSOLE_LOG_FREQ = "console_log_freq"
+    CHECKPOINT_FREQ = "checkpoint_freq"
+    BATCH_SIZE = "batch_size"
+
+
+# parameters relevant for heterogeneous structure
+class HeteroParams:
+    FEATURES_NAME = "features_name"
+    OBSERVED_ATTRIBUTE = "observed_attribute"
+    CLASS_NAME = "class_name"
+    REINDEXING = "reindexing"
+    INV_REINDEXING = "inv_reindexing"
+
+
+# other necessary parameters
+class OtherParams:
+    DEVICE_TYPE = "device_type"
+    PATH_TO_MODEL = "path_to_model"
+    PATH_TO_MODEL_LAST = "path_to_model_last"
+    PATH_TO_MODEL_SECOND_LAST = "path_to_model_second_last"
+    PATH_TO_MODEL_THIRD_LAST = "path_to_model_third_last"
+
+
+# main variables for modelling
+class Modelling:
+    # all None until set_params are executed
+    data: mgp.Any = None
+    # model: InductiveModel = None
+    model: mgp.Any = None
+    opt = None
+    criterion = None
```

Comment:
I would remove this from Modeling, and specify them better with Typing. These can be global without having this class

Reply:
Removed class Modelling. Discuss typing on meeting.